lib/response.js

```diff
@@ -463,7 +463,9 @@ module.exports = {
       this.res.setHeader(field, val);
     } else {
       for (const key in field) {
-        this.set(key, field[key]);
+        if (Object.prototype.hasOwnProperty.call(field, key)) {
```

Comment:
It can be a breaking change, why you need to add this check?

Reply:
Hello, the reason I want to modify is: checks for unfiltered for-in loops, The use of this construct results in processing inherited or unexpected properties. Make the code more robust.

Reply:
Sorry, if there is no actual scene, we will not modify it here, because it runs the risk of breaking other people's code

Reply:
Got it, thank you